src/shared/ThisContext.js

```diff
@@ -180,21 +182,52 @@ export default class UIHContext {
    * @param {object} value:  An object containing {prop:value} to add
    */
   props(value) {
+    let _value = value;
     // WRITE.
-    if (R.is(Object, value)) {
+    if (R.is(Object, _value)) {
       // Cumulatively add given props to the existing
       // props on the component.
-      const component = api.component();
-      let props = component && component.props;
-      if (props) {
-        props = R.clone(props);
-        R.keys(value).forEach(key => props[key] = value[key]);
-        value = props;
-      }
+      const component = this[PROP]('componentProps');
+      const props = (component && component.props) || {};
+      // No need to clone when using R.merge
+      _value = R.merge(props, _value);
     }
-
     // READ.
-    return this[PROP]('componentProps', value);
+    return this[PROP]('componentProps', _value);
+  }
+
```

Comment:
Let's add a method like this as a temporary holding pattern for all the specs that are going to break out there in the world (not least of which are mine! :smile: ):

```js
  /**
   * OBSOLETE: Stub for the `component` method, emitting a warning that it will
   * be deprecated in a future version.
   */
  load(component) {
    log.warn('The "load" method is deprecated. Please use the "component" method.');
    return this.component(component)
  }
```

![image](https://cloud.githubusercontent.com/assets/185555/13208892/7ed00022-d982-11e5-9ba7-fd09c8702a9f.png)

This will require the following import at the head of the file:

```js
import log from './log';
```